pkg/util/ecs/common.go

```diff
@@ -26,22 +27,7 @@ const (
 
 // GetTaskMetadata extracts the metadata payload for the task the agent is in.
 func GetTaskMetadata() (TaskMetadata, error) {
-	var meta TaskMetadata
-	client := http.Client{
-		Timeout: timeout,
-	}
-	resp, err := client.Get(metadataURL)
-	if err != nil {
-		return meta, err
-	}
-	defer resp.Body.Close()
-
-	decoder := json.NewDecoder(resp.Body)
-	err = decoder.Decode(&meta)
-	if err != nil {
-		log.Errorf("decoding task metadata failed - %s", err)
-	}
-	return meta, err
+	return getTaskMetadataWithURL(metadataURL)
```

Comment:
Just want to clarify something to be sure my understanding is correct: this information will only be collected on Fargate, right? If so, I think it would be good to mention this in the PR title/description.

Reply:
Thank you for mentioning that, I updated the PR's title and created a card to collect network info from ECS EC2 launch-type-based containers which will require a totally different logic compared to this PR